### PR TITLE
Add phpstan and psalm

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this change respect BC.

### CODE REVIEW

**Review by VincentLanglet — COMMENTED** [review]
A PR is needed on dev-kit

**Comment by wbloszyk** [comment]
> A PR is needed on dev-kit

I know. A coffee break.

https://github.com/sonata-project/dev-kit/pull/1694/files

File: src/Admin/PostAdmin.php [hunk 1]
```diff
@@ -38,7 +37,7 @@ class PostAdmin extends AbstractAdmin
     /**
      * @deprecated since sonata-project/news-bundle 3.13, to be removed in 4.0.
      *
-     * @var UserManagerInterface|null
+     * @var null
```
Comment (VincentLanglet) [inline src/Admin/PostAdmin.php]: I assume it's not always null ?

File: src/Admin/PostAdmin.php [hunk 2]
```diff
@@ -55,7 +54,7 @@ class PostAdmin extends AbstractAdmin
     /**
      * @deprecated since sonata-project/news-bundle 3.13, to be removed in 4.0.
      *
-     * @param UserManagerInterface|null $userManager
+     * @param null $userManager
```
Comment (VincentLanglet) [inline src/Admin/PostAdmin.php]: I assume it's not always null ?

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/Admin/PostAdmin.php (hunk 1 above)
Comment (wbloszyk) [reply src/Admin/PostAdmin.php]: This property is not used. Keeping `UserManagerInterface` it is not nesessery.

Comment (wbloszyk) [reply src/Admin/PostAdmin.php]:
```suggestion
```

File: src/Admin/PostAdmin.php (hunk 2 above)
Comment (wbloszyk) [reply src/Admin/PostAdmin.php]:
```suggestion
```

**Review by VincentLanglet — DISMISSED** [review]

File: src/Document/PostManager.php [hunk 3]
```diff
@@ -23,10 +24,10 @@
...
 {
     /**
-     * @param $year
-     * @param $month
-     * @param $day
-     * @param $slug
+     * @param string $year
+     * @param string $month
+     * @param string $day
+     * @param string$slug
```
Comment (wbloszyk) [inline src/Document/PostManager.php]:
```suggestion
     * @param string $slug
```

**Review by VincentLanglet — DISMISSED** [review]

File: composer.json [hunk 4]
```diff
@@ -53,6 +53,7 @@
     },
     "conflict": {
         "doctrine/dbal": "<2.6.0",
+        "doctrine/mongodb-odm": "<2.0",
```
Comment (core23) [inline composer.json:56]:
```suggestion
        "doctrine/mongodb-odm": "<2.2",
```

Comment (core23) [reply composer.json:56]: This should match the require-dev dependency

File: src/Command/SynchronizeCommentsCountCommand.php [hunk 5]
```diff
@@ -28,6 +28,9 @@ public function configure()
         $this->setDescription('Synchronize comments count');
     }
 
+    /**
+     * @return void
+     */
     public function execute(InputInterface $input, OutputInterface $output)
```
Comment (core23) [inline src/Command/SynchronizeCommentsCountCommand.php:34]: Commands shoult return an int value

**Review by core23 — CHANGES_REQUESTED** [review]

File: src/Command/SynchronizeCommentsCountCommand.php (hunk 5 above)
Comment (wbloszyk) [reply src/Command/SynchronizeCommentsCountCommand.php:34]: Done.

File: composer.json (hunk 4 above)
Comment (wbloszyk) [reply composer.json:56]: Done.

**Review by VincentLanglet — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**